### PR TITLE
blockchain: added more info to preCheck and buyGas error

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- When an error is returned from `preCheck` or `buyGas` function, it only returns the type of the error, not the information related to the error
- To help debugging, wrapping an error before returning the error to the caller

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by KimKyungup** [comment]
I think we can wrap errors like below.
https://blog.golang.org/go1.13-errors

**Comment by KimKyungup** [comment]
I think some clients check if the error is same with `ErrNonceTooLow` or `ErrNonceTooHigh.`
This change may not be incompatible with the clients.

locustSlave do this. Maybe Caver??
@jimni1222 Is there any code to check if the error is same with them in Caver?

**Comment by jimni1222** [comment]
> I think some clients check if the error is same with `ErrNonceTooLow` or `ErrNonceTooHigh.`
> This change may not be incompatible with the clients.
> 
> locustSlave do this. Maybe Caver??
> @jimni1222 Is there any code to check if the error is same with them in Caver?

Yes, in caver some tests check ErrTooLow.
If platform changes error message, i can edit test after release.

**Comment by ehnuje** [comment]
Hmm, things become more complicated. How about just leaving a log instead of returning the wrapped error? @KimKyungup @jimni1222

**Comment by KimKyungup** [comment]
> Hmm, things become more complicated. How about just leaving a log instead of returning the wrapped error? @KimKyungup @jimni1222

I agree with you. Just leave a log.

**Comment by ehnuje** [comment]
Dear reviewers, could you please review this change? @winnie-byun @kjhman21 @KimKyungup @aidan-kwon

File: blockchain/state_transition.go [hunk 1]
```diff
@@ -170,21 +170,30 @@ func (st *StateTransition) buyGas() error {
 	validatedSender := st.msg.ValidatedSender()
 	feeRatio, isRatioTx := st.msg.FeeRatio()
 	if isRatioTx {
-		feePayer, feeSender := types.CalcFeeWithRatio(feeRatio, mgval)
+		feePayerFee, senderFee := types.CalcFeeWithRatio(feeRatio, mgval)
 
-		if st.state.GetBalance(validatedFeePayer).Cmp(feePayer) < 0 {
+		if st.state.GetBalance(validatedFeePayer).Cmp(feePayerFee) < 0 {
+			logger.Error("%w - feePayer: %v, feePayerBalance: %v, feePayerFee: %v, txHash: %v",
```
Comment (KimKyungup) [inline blockchain/state_transition.go:176]: How about Debug log level? I think this is not an error case of Klaytn node.

And how about logging like this?

```suggestion
			logger.Debug("StateTransition buyGas failed", "err"r, errInsufficientBalanceForGasFeePayer, "feePayer",  validatedFeePayer.String() ,"feePayerBalance", st.state.GetBalance(validatedFeePayer).Uint64()...
```

**Comment by aidan-kwon** [comment]
@ehnuje `preCheck` and `buyGas` can return errors in benign cases when miner's `commitNewWork()` is executed faster than txpool's `reset()`. I think error level logs are inappropriate in those cases.

**Comment by ehnuje** [comment]
@KimKyungup @aidan-kwon I agree with your opinion, updated!

File: blockchain/state_transition.go [hunk 2]
```diff
@@ -170,21 +170,30 @@ func (st *StateTransition) buyGas() error {
 	validatedSender := st.msg.ValidatedSender()
 	feeRatio, isRatioTx := st.msg.FeeRatio()
 	if isRatioTx {
-		feePayer, feeSender := types.CalcFeeWithRatio(feeRatio, mgval)
+		feePayerFee, senderFee := types.CalcFeeWithRatio(feeRatio, mgval)
 
-		if st.state.GetBalance(validatedFeePayer).Cmp(feePayer) < 0 {
+		if st.state.GetBalance(validatedFeePayer).Cmp(feePayerFee) < 0 {
+			logger.Debug("%w - feePayer: %v, feePayerBalance: %v, feePayerFee: %v, txHash: %v",
```
Comment (KimKyungup) [inline blockchain/state_transition.go:176]: `logger` does not support this style.
`logger.Debug("blah %v", feePayerFee)`

Please write this sytle.
`logger.Debug("blah ", "feePayerFee",  feePayerFee)`

Comment (ehnuje) [reply blockchain/state_transition.go]: Oh, I forgot to change the log from string to key value pairs while changing it from an error to the log. Thanks for checking the issue!

**Pull request merged**